### PR TITLE
feat(source-iterable): introduce EU API endpoint

### PR DESCRIPTION
**Pull request by JaydanPB**
## What
Resolving https://github.com/airbytehq/airbyte/issues/39488 by introducing a required region parameter.

## How
Introduces a new parameter in the connector specification called region. This region parameter is used to determine the API URL:
```
{{ 'https://api.eu.iterable.com/api/' if
config.get('region', "US") == 'EU' else
'https://api.iterable.com/api/' }}
```

## User Impact
Users will be required to pick a region (US or EU) during the setup of their Iterable source. 

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-iterable/source_iterable/source.py [hunk 1]
```diff
@@ -66,6 +66,7 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         # end date is provided for integration tests only
         start_date, end_date = config["start_date"], config.get("end_date")
         date_range = {"start_date": start_date, "end_date": end_date}
+        region = config["region"]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-iterable/source_iterable/source.py:69]:
```suggestion
        region = config["region"] or "US"
```

This way it won't break existing connections.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments.

**Comment by marcosmarxm** [comment]
Please sign the CLA.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
LGTM @JaydanPB can you bump the connector version inside the metadata.yaml and pyproject.toml file. Also add the entry for the changelog in documentation

**Comment by JaydanPB** [comment]
This has now been tested - it successfully synced data from Iterable EU as expected.  

I do not have access to a Iterable US API key so I cannot fully test whether we have introduced any breaking changes, but I have attempted to add the region parameter in the least intrusive way possible and a rollback should be straightforward.

**Pull request closed without merging**